clients/src/main/java/org/apache/kafka/clients/consumer/internals/CommitRequestManager.java

```diff
@@ -425,6 +427,13 @@ public CompletableFuture<Map<TopicPartition, OffsetAndMetadata>> commitSync(fina
         CompletableFuture<Map<TopicPartition, OffsetAndMetadata>> result = new CompletableFuture<>();
         OffsetCommitRequestState requestState = createOffsetCommitRequest(commitOffsets, deadlineMs);
         commitSyncWithRetries(requestState, result);
+        new ArrayList<>(metadataErrors).forEach(metadataError -> metadataError.whenComplete((__, error) -> {
+            if (error != null) {
+                result.completeExceptionally(error);
+                metadataErrors.remove(metadataError);
+            }
+        }));
+        
```

Comment:
this is only handling the commit sync but we do have other requests that should also fail if there is a fatal error in the coordinator. Given that all requests that this manager sends end up being added to the `pendingRequests`, and we attempt to send them on poll, shouldn't we just check on poll of there is no coordinator due to a fatal error and fail all pendingRequests? It would ensure we behave consistently for all requests.

```
    public NetworkClientDelegate.PollResult poll(final long currentTimeMs) {
        if (!coordinatorRequestManager.coordinator().isPresent()) {
            if (coordinatorRequestManager.fatalError().isPresent()) {
                // Fatal error looking up coordinator => Fail all pendingRequests
            }
            return EMPTY;
        }
        ...
```
I created this separate Jira https://issues.apache.org/jira/browse/KAFKA-18034 for this, given that it's a different gap and fix. I would suggest we address that in a separate/simpler PR with these thoughts, what do you think?

Reply:
Yes, I think we should deal this fataError at another PR, it is a little bit different with this PR, because this PR should focus on Error Event how to passing to application thread.

Reply:
thanks for considering the split! Sounds good.